cmd/zpool/zpool_main.c

```diff
@@ -3761,9 +3761,12 @@ print_vdev_stats(zpool_handle_t *zhp, const char *name, nvlist_t *oldnv,
 		return (ret);
 
 	if (oldnv && nvlist_lookup_nvlist_array(oldnv, ZPOOL_CONFIG_CHILDREN,
-	    &oldchild, &c) != 0)
+	    &oldchild, &oldchildren) != 0)
 		return (ret);
 
+	if (oldnv != NULL)
+		children = MIN(oldchildren, children);
+
```

Comment:
The fix looks good, but what do you think about restructuring it like this:

```c
        if (oldnv) {
                if (nvlist_lookup_nvlist_array(oldnv, ZPOOL_CONFIG_CHILDREN,
                    &oldchild, &oldchildren) != 0)
                        return (ret);

                children = MIN(oldchildren, children);
        }
```

Reply:
Yea, that's better.  I just included it in my latest push.